spec/lang/representation.md

```diff
@@ -405,22 +402,157 @@ Unsized types do not have values and thus there is no representation relation.
 ```rust
 impl Type {
     fn decode<M: Memory>(Type::Slice { .. }: Self, bytes: List<AbstractByte<M::Provenance>>) -> Option<Value<M>> {
-        panic!("tried to decode a slice")
+        panic!("decode of Type::Slice")
     }
     fn encode<M: Memory>(Type::Slice { .. }: Self, val: Value<M>) -> List<AbstractByte<M::Provenance>> {
-        panic!("tried to endoce a slice")
+        panic!("encode of Type::Slice")
+    }
+}
+```
+
+## Well-formed values
+
+We call a value `val` *well-formed* for a type `ty` if `machine.check_value(val, ty).is_ok()`.
+The specification ensures (statically or dynamically) that all values in MiniRust are always well-formed, the only exception being values returned from `decode`, which are sanitized by [typed memory accesses](#Typed-memory-accesses).
+Violating this when running a well-formed program is a spec bug.
+Therefore `encode` and other expressions can assume well-formed inputs as well.
+
+In particular must a load instruction cause UB, when the sequence of bytes does not correspond to any representation of a well-formed value.
+Generally, this is already dealt with by `decode`: if decoding succeeds, the representation is well-formed.
+However, pointer types have further runtime constraints: specifically, [safe pointers][ptr_type] require the address to be aligned, dereferenceable and non-null, and they require the pointee type to be inhabited.
+This is separated from `decode`, because checking alignment or the well-formedness of a trait object pointer needs vtable information, and dereferenceablity need the allocations.
```

Comment:
```suggestion
This is separated from `decode`, because checking alignment or the well-formedness of a trait object pointer needs vtable information, and dereferenceablity need information about the current state of memory.
```